app/main.py

```diff
@@ -1,2 +1,51 @@
+from dataclasses import dataclass
+from typing import Any, Hashable, Optional
+
+
 class Dictionary:
-    pass
+    @dataclass
+    class Node:
+        key: Hashable
```

Comment:
Hash should be also included here as per task requirements - we would use it later to check if hashes are matching

Reply:
You might not need to check hash in your implementation but I would still recommend to store the hash in a Node to add an extra check that both key and hash are matching when we are going through the indexes of hash table. Just in case, to be 100% sure that we get correct value.